components/ActionSheet/src/MDCActionSheetController.m

```diff
@@ -0,0 +1,331 @@
+  CGRect tableFrame = _tableView.tableView.frame;
+  tableFrame.origin.y = CGRectGetHeight(self.header.frame);
+  _tableView.tableView.frame = tableFrame;
+  self.preferredContentSize = CGSizeMake(width, height);
+  initialLayout = true;
+}
+
+- (void)viewWillAppear:(BOOL)animated {
+  [super viewWillAppear:animated];
+
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
+  self.mdc_bottomSheetPresentationController.delegate = self;
+#pragma clang diagnostic pop
+
+  self.mdc_bottomSheetPresentationController.dismissOnBackgroundTap =
+      _transitionController.dismissOnBackgroundTap;
+  [self.view layoutIfNeeded];
+}
+
+- (BOOL)accessibilityPerformEscape {
+  if (!self.dismissOnBackgroundTap) {
+    return NO;
+  }
+  [self dismissViewControllerAnimated:YES completion:nil];
+  return YES;
+}
+
+- (void)preferredContentSizeDidChangeForChildContentContainer:(id<UIContentContainer>)container {
+  [super preferredContentSizeDidChangeForChildContentContainer:container];
+  [self.presentationController
+      preferredContentSizeDidChangeForChildContentContainer:self];
+}
+
+- (UIScrollView *)trackingScrollView {
+  return _transitionController.trackingScrollView;
+}
+
+- (void)setTrackingScrollView:(UIScrollView *)trackingScrollView {
+  _transitionController.trackingScrollView = trackingScrollView;
+}
+
+- (BOOL)dismissOnBackgroundTap {
+  return _transitionController.dismissOnBackgroundTap;
+}
+
+-(void)setDismissOnBackgroundTap:(BOOL)dismissOnBackgroundTap {
+  _transitionController.dismissOnBackgroundTap = dismissOnBackgroundTap;
+  self.mdc_bottomSheetPresentationController.dismissOnBackgroundTap = dismissOnBackgroundTap;
+}
+
+- (void)viewWillTransitionToSize:(CGSize)size withTransitionCoordinator:
+    (id<UIViewControllerTransitionCoordinator>)coordinator {
+  [super viewWillTransitionToSize:size withTransitionCoordinator:coordinator];
+  CGRect frame = self.view.frame;
+  frame.size = size;
+  self.view.frame = frame;
+  CGRect headerFrame = self.header.frame;
+  headerFrame.size.width = size.width;
+  self.header.frame = headerFrame;
+  [self.header setNeedsLayout];
+  [self.header layoutIfNeeded];
+
+  CGFloat height = CGRectGetHeight(self.header.frame) + [_tableView tableHeightForWidth:size.width];
+  CGSize updatedSize = CGSizeMake(size.width, height);
+  self.preferredContentSize = updatedSize;
+}
+
+-(void)setTransitioningDelegate:(id<UIViewControllerTransitioningDelegate>)transitioningDelegate {
+  NSAssert(NO, @"MDCActionSheetController.transitionDelegate cannot be changed");
+  return;
+}
+
+-(void)setModalPresentationStyle:(__unused UIModalPresentationStyle)modalPresentationStyle {
+  NSAssert(NO, @"MDCActionSheetController.modalPresentationStyle cannot be changed.");
+  return;
+}
+
+- (MDCActionSheetHeaderView *)headerView {
+  CGRect headerRect = CGRectZero;
+  headerRect.size.width = CGRectGetWidth(self.view.frame);
```

Comment:
When calculating frames for subviews you must use the parent view's `bounds` rather than its frame.

`frame` is a composite API that represents the view's `layer.position`, `layer.bounds`, `layer.transform`, and `layer.anchorPoint`. You want to know the dimensions of the parent view, which is strictly defined by layer.bounds (or its mirror API view.bounds), so you want to use view.bounds directly.

Reply:
Done